src/modules/web3/services/EthProvidersService.ts

```diff
@@ -99,8 +99,13 @@ export class EthProvidersService {
 
     if (!transaction) {
       const web3Transaction = await this.getProvider(chainId).getTransaction(hash);
-      const { data } = web3Transaction;
-      transaction = this.transactionRepository.create({ chainId, data, hash });
+      const { data, blockNumber } = web3Transaction;
```

Comment:
Are we planning to use this database for other services? The only reason I ask is that the stored data is somewhat limited for each of the types (transactions, blocks, especially), and I wanted to better understand how painful the upgrade process will be once this is in production and we decide that we want to add a new transaction field to the db (for instance).

Put differently, will we be building a separate transaction, block, and event database for each service that we run (referrals, across API, transactions page)? Or will we be building a unified API that serves all of them through the same DB that is easy to upgrade when we need a new field.

Note: this may just be my ignorance about how SQL databases are typically managed.

Reply:
@mrice32 It is a unified API that will use the same tables. The table fields are easy to update at a later time 👍